src/utils/DependencyUtils.ts

```diff
@@ -0,0 +1,9 @@
+export const splitToNameAndVersion = (dependencyString: string): [string, string] => {
+    const parts = dependencyString.split('-');
+
+    if (parts.length !== 3) {
+        throw new Error(`Invalid dependency string "${dependencyString}"`);
```

Comment:
Careful, this is invalid. Author names can have hyphens iirc.

You can assume the dependency string **IS** valid because we're supplied it by TS. If a user attempts to import a local mod and gets it wrong then we just won't resolve it.

You can instead do (pseudo):
```
splits = dependencyString.split('-');
splits.last() => version
splits.secondLast() => mod name
splits.rest().join("-") => author name
```

Reply:
Changed the implementation to [accept more than two hyphens](https://github.com/ebkr/r2modmanPlus/pull/1482/commits/9d71ae7c907b9b0ca24911a070847b4cc03b4974#diff-485ac28d8d642a9d12bce5c6751707214cc35527c52eac5c156d29640449b882R1).